/* XPECT_SETUP org.uqbar.project.wollok.tests.xpect.WollokXPectTest END_SETUP */

class A {
	method inABlock() {
		const aBlock = {a =>
			throw new Exception()
			// XPECT warnings --> "Unexpected expression after throw" at "a > 23"
			a > 23
		}
		aBlock.apply(23)
		throw new Exception()
		
		// XPECT warnings --> "Unexpected expression after throw" at "aBlock.apply(23)"
		aBlock.apply(23)
		// XPECT warnings --> "Unexpected expression after throw" at "return '23'"
		return '23'
	}
	
	method withinAnIf(aParam) {
		var a = 1
		if (a > 0) {
			throw new Exception()
			// XPECT warnings --> "Unexpected expression after throw" at "a = 23"
			a = 23
		}
		else {
			a = 22
			throw new Exception()
			// XPECT warnings --> "Unexpected expression after throw" at "a = 21"
			a = 21
		}
	}
	
	method throwAfterThrow() {
		throw new Exception()
		// XPECT warnings --> "Unexpected expression after throw" at "throw new Exception()"
		throw new Exception()
	}
}
